commit f6586a878bc4e090dec3546c5e49107990e26eaf
Author: Eli Cohen <eli@mellanox.co.il>
Date:   Sun Aug 29 16:45:21 2010 +0300

    mlx4: Add a new vendor ID method to mlx4
    
    Make method with attribute ID 0xff70 be proccessed by FW.
    
    Signed-off-by: Eli Cohen <eli@mellanox.co.il>

Index: ofed_kernel-fixes/drivers/infiniband/hw/mlx4/mad.c
===================================================================
--- ofed_kernel-fixes.orig/drivers/infiniband/hw/mlx4/mad.c	2010-10-14 08:56:00.000000000 +0200
+++ ofed_kernel-fixes/drivers/infiniband/hw/mlx4/mad.c	2010-10-14 09:33:53.188193713 +0200
@@ -229,6 +229,16 @@ static void forward_trap(struct mlx4_ib_
 	}
 }
 
+static int is_vendor_id(__be16 attr_id)
+{
+	return (attr_id & IB_SMP_ATTR_VENDOR_MASK) == IB_SMP_ATTR_VENDOR_MASK;
+}
+
+static int supported_vendor_id(__be16 attr_id)
+{
+	return 1;
+}
+
 static int ib_process_mad(struct ib_device *ibdev, int mad_flags, u8 port_num,
                           struct ib_wc *in_wc, struct ib_grh *in_grh,
                           struct ib_mad *in_mad, struct ib_mad *out_mad)
@@ -256,8 +266,8 @@ static int ib_process_mad(struct ib_devi
 		 * MADs -- the SMA can't handle them.
 		 */
 		if (in_mad->mad_hdr.attr_id == IB_SMP_ATTR_SM_INFO ||
-		    ((in_mad->mad_hdr.attr_id & IB_SMP_ATTR_VENDOR_MASK) ==
-		     IB_SMP_ATTR_VENDOR_MASK))
+		    (is_vendor_id(in_mad->mad_hdr.attr_id) &&
+		    !supported_vendor_id(in_mad->mad_hdr.attr_id)))
 			return IB_MAD_RESULT_SUCCESS;
 	} else if (in_mad->mad_hdr.mgmt_class == IB_MGMT_CLASS_PERF_MGMT ||
 		   in_mad->mad_hdr.mgmt_class == MLX4_IB_VENDOR_CLASS1   ||
